org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -239,10 +249,15 @@ private void appendGuessingCompletion(StringBuilder buffer, CompletionProposal p
 			}
 
 			char[] argument = parameterNames[i];
-
-			buffer.append("{{");
+			if(supportSnippets){
+				buffer.append("${");
+				buffer.append(Integer.toString(i+1));
+				buffer.append(":");
+			}
 			buffer.append(argument);
```

Comment:
I guess what I'm saying is that until #12, when the client doesn't support `SnippetString` API, it doesn't seem right to return any parameters at all, because that's pretty much never what the user wants to insert. I think it is best to insert the method name and have them continue typing, rather than inserting `method name + ( + name of parameter.... + ) + ;` and then ask her to go and delete the extraneous stuff. Otherwise i will have to post-process the completion items and regex it out, which isn't super fun or robust.

Are there other clients which use/want this insertion string? 

In case i'm not being clear, here's what currently happens with this PR:

![ycm-java-language-sever-completion](https://cloud.githubusercontent.com/assets/10584846/23578744/b9065cb6-00d5-11e7-91d9-0f6d61630579.gif)